core/src/main/java/forklift/consumer/Consumer.java

```diff
@@ -483,10 +405,154 @@ else if (producer.topic().length() > 0)
         return closeMe;
     }
 
+    private Object constructMessageHandlerInstance(ForkliftMessage forkliftMessage, List<Closeable> closeables) throws IllegalAccessException, InvocationTargetException, InstantiationException, IOException {
+        Object instance = null;
+        if (this.constructor != null) {
+            Object[] constructorParameters = buildConstructorParameters(forkliftMessage, closeables);
+            instance = this.constructor.newInstance(constructorParameters);
+        } else {
+            instance = msgHandler.newInstance();
+        }
+        return instance;
+    }
+
+    private Object[] buildConstructorParameters(ForkliftMessage forkliftMessage, List<Closeable> closeables) throws IOException {
+        Object[] parameters = new Object[constructorAnnotations.length];
+        int index = 0;
+        for (Annotation[] parameterAnnotations : constructorAnnotations) {
+            //there must be at least one annotation that indicates what we should inject
+            Annotation injectable = null;
+            for (Annotation parameterAnnotation : parameterAnnotations) {
+                if (injectFields.containsKey(parameterAnnotation.annotationType())) {
+                    injectable = parameterAnnotation;
+                    break;
+                }
+            }
+            Parameter p = constructor.getParameters()[index];
+            Object value = getInjectableValue(injectable, null, p.getType(), forkliftMessage);
+            parameters[index] = value;
+            if (value != null && value instanceof ForkliftProducerI) {
+                closeables.add((ForkliftProducerI)value);
+            }
+
+            index++;
+        }
+        return parameters;
+    }
+
+    private Object getInjectableValue(Annotation decorator, String mappedName, Class<?> mappedClass, ForkliftMessage msg) throws IOException {
+        Object value = null;
+        if (decorator == null || decorator.annotationType() == javax.inject.Inject.class) {
+            if (this.services != null) {
+                // Try to resolve the class from any available BeanResolvers.
+                for (ConsumerService s : this.services) {
+                    try {
+                        final Object o = s.resolve(mappedClass, null);
+                        if (o != null) {
+                            value = o;
+                            break;
+                        }
+                    } catch (Exception e) {
+                        log.debug("", e);
```

Comment:
I know this was existing code, but aren't errors like "failed to autowire dependency" pretty useful?